File: Find Valid Matrix Given Row and Column Sums.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        n = len(rowSum)
        m = len(colSum)
        matrix = [[0] * m for _ in range(n)]
        i = 0
        j = 0
        while i < n and j < m:
            matrix[i][j] = min(rowSum[i], colSum[j])
            rowSum[i] -= matrix[i][j]
            colSum[j] -= matrix[i][j]
            while i < n and rowSum[i] == 0:
                i += 1

            while j < m and colSum[j] == 0:
                j += 1
        return matrix

File: Find Valid Matrix Given Row and Column Sums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        int n = rowSum.size(), m = colSum.size();
        vector<vector<int>> matrix;
        matrix.reserve(n);
        for (int i = 0; i < n; i++) {
            matrix.emplace_back(m, 0);
        }
        int i = 0, j = 0;
        while (i < n && j < m) {
            matrix[i][j] = min(rowSum[i], colSum[j]);
            rowSum[i] -= matrix[i][j];
            colSum[j] -= matrix[i][j];
            while (i < n && rowSum[i] == 0) i++;
            while (j < m && colSum[j] == 0) j++;
        }
        return matrix;
    }
};
